src/fparser/two/tests/fortran2003/test_specific_binding_r451.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (c) 2018 Science and Technology Facilities Council
+
+# All rights reserved.
+
+# Modifications made as part of the fparser project are distributed
+# under the following license:
+
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+
+# 1. Redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer.
+
+# 2. Redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution.
+
+# 3. Neither the name of the copyright holder nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+'''
+Test Fortran 2003 rule R451 : This file tests the support for specific
+binding and passed-object dummy argument within the type-bound procedure
+part of a derived type.
+'''
+
+import pytest
+from fparser.two.utils import NoMatchError, InternalError
+from fparser.two.Fortran2003 import Specific_Binding
+
+
+def test_valid(f2003_create):
+    '''Test that valid Specific_Binding statements are parsed
+    correctly.
+
+    '''
+
+    # simple statement
+    obj = Specific_Binding('procedure sub')
+    assert str(obj) == "PROCEDURE sub"
+
+    # simple statement no spaces
+    obj = Specific_Binding('proceduresub')
```

Comment:
Is that really correct? Is sub a key word or the name of the procedure/routine?

Reply:
I agree it is not what one might expect but it works in gfortran so I kept it in. There does not seem to be any guidance in the spec as far as I can tell.

The example is the keyword "procedure" followed by the procedure/routine name "sub".

I can disallow if you like.

Reply:
OK, the deed is done. This may come back to bite us one day.